[TextInput] Add new "masked" prop to obscure hidden data

nkohari opened the pull request:
As described in #185, the use of `type=password` to hide non-password information triggers password managers like 1Password. This patch adds a prop called `masked` to the `TextInput` which will manually obscure the value, and force the input to be `readOnly`.

If we decide to go with this proposal, we should definitely test switching a managed `TextInput` between masked and "not-masked" to make sure the value isn't lost. If the component is managed (that is, the`value` comes from a parent component) it should be okay.

WDYT?

landitus commented:
I like the idea! In the docs I only see the field turned into a read-only, but the value is still visible. Maybe it's a matter of replacing the characters with the circle symbol to mimic the password field output: https://stackoverflow.com/a/37965943/272553

nkohari commented:
I changed this slightly to use the `placeholder` property instead of `defaultValue`, and to null out any value that would normally be passed.

This means that the value of the `<input>` will actually be `null` if you read it from the DOM element itself, but this should be okay as long as the component isn't intended to support "regular" HTML forms. (If the `<form>` the `<TextInput>` is in is submitted, the form body won't include the value for masked `<TextInput>`s, because they've been set to null.)

I also added an example in the docs to illustrate how this works. @siddharthkp, WDYT about this?
